environment.yml

```diff
@@ -1,54 +1,97 @@
+# environment.yml - conda environment specification file for RMG-Py
+#
+# Usage: conda env create --file environment.yml
+#
+# This file contains all of the software packages needed to run RMG-Py.
+# There is a mixture of the following
+# - packlages would could be installed at OS level, but we install here
+#   for better version control
+# - python tools
+# - external software tools specific to chemistry
+# - other software we maintain which RMG depends on
+# + some other categories (see below)
+#
+# Changelog:
+# - May 15, 2023 Added this changelog, added inline documentation,
```

Comment:
The way I see changelogs is that they provide a "high-level" perspective on how the file has changed over time. The git log is useful for fixing technical issues and precise problems with specific lines, but I think the changelog is better for answering questions like "why do we have some packages from this channel but other from this one" or the like.

`environment.yml` in particular has been going through substantial changes lately, and as we (hopefully) move toward using fewer and fewer in-house packages I think it will be useful to have this high level history of that activity. This is also a file that community developers will interact with a lot - such info would be particularly useful for them.